docs/source/includes/_translations.md

```diff
@@ -0,0 +1,178 @@
+            "type": "tutorials"
+        },
+        "translations.workflow": {
+            "href": "/workflows/{translations.workflow}",
+            "type": "workflows"
+        }
+    }
+}
+
+```
+
+foo
+## `strings` and `string_versions`
+
+Whenever a collaborator changes translatable resources, Panoptes will copy over
+those strings into the "primary" translation resource. This is the translation
+resource that has its language set to the language of the translatable resource
+(which it will typically inherit from the Project it belongs to, where
+applicable). This primary translation resource is not editable via the API.
+
+Upon copying data into the primary translation resource, Panoptes calculates a
+version number for the translated resource. This version number will be
+applicable to the strings that were changed. For instance, if a project
+translation is initially this:
+
+<div class="center-column"></div>
+```json
+{
+  "strings": {"title": "Hello", "description": "This needs to be written"},
+  "string_versions": {"title": 1, "description": 1}
+}
+```
+
+And the project owner updates the description (via the [Projects](#projects)
+API) but not the title, once Panoptes finishes updating the translation (which
+happens asynchronously), the translation resource will look like this:
+
+<div class="center-column"></div>
+```json
+{
+  "strings": {"title": "Hello", "description": "This is now a real description"},
+  "string_versions": {"title": 1, "description": 2}
+}
+```
+
+Any translation editing UI should use this to provide the following features:
+
+1. If for instance the string version of `description` on a secondary
+   translation is set to `1`, while the primary translation has a version number
+   of `2` or higher for this field, in the UI this field should be marked as
+   outdated to let translators know that the project owner has updated the
+   description and they need to recheck and update the translation for this
+   field.
+0. When a translator updates a field, the UI should send along the number for
+   that field's string version of the primary translation, as shown to the
+   translator. That is to say, if a translator then updates the `description`
+   field of the secondary translation, when the editing UI calls the Panoptes
+   API to save the updated translation, it should send along the value `2` for
+   the `string_versions["description"]`. It should only update the values for
+   strings that were actually touched, so as no to reset the outdatedness
+   information for fields that weren't updated yet.
+
+
+## List translations
+
+## Get a single translation
+
+## Create a translation
+
+## Update a translation
+
+## Publish the latest translation
+
+```http
+POST /api/translations/123/publish HTTP/1.1
+Accept: application/vnd.api+json; version=1
+Content-Type: application/json
+```
+
+This API will mark the latest version of this translation as the published version.
```

Comment:
What does 'latest version of this translation' mean here if the translation's strings have mixed version numbers eg. `{  "title": 1, "description": 2 }`?

Reply:
Translation string versions relate to the translated resource, i.e. a `project` has `project_versions` 1 and 2, the title attribute hasn't changed since version 1, description was updated in version 2. 

Publishing a translation locks the set of translated strings and their resource version ids to the current known values. 

E.g. with the following translation data, `project_version` 2 contains both strings, `project_version` 1 i missing the updated description.
```
string_versions = { "title": 1, "description": 2 },
strings = { "title": "An Amazing Title", "description": "Descriptive descriptions delight" }
```

Publishing that translation locks the `translation.published_version` to the above. A translator could update the translations but they won't make it to the live site till those changes are published.

Reply:
I'm more confused now about how this works. So the published version can be a partially updated translation? If the translator later updates the title to version 2, is that published to a new version since the published version 2 can't be updated?

I'm assuming that the original has `string_versions: { "title": 2, "description": 2 }` but the translation is `string_versions: { "title": 2, "description": 2 }`.

Reply:
> So the published version can be a partially updated translation?

Yeah - it'll contain anything the translation strings are set to from pandora, which can be an outdated / partially translated set of strings. 

> If the translator later updates the title to version 2, is that published to a new version since the published version 2 can't be updated?

The changes to title are stored on the `translation` resource and recorded in a `translation_version` as well. Translation versions don't change they are append only and contain the updated strings between translations. Published versions just point to a different version in time. 

The new strings won't be published by default. When a project owner chooses to publish the new strings, the translation resource updates it's `published_version` link to point to the latest version. 

> I'm assuming that the original has string_versions: { "title": 2, "description": 2 } but the translation is string_versions: { "title": 1, "description": 2 }

The original being the primary language? Yeah it'll always have the latest string versions. And yep the translation itself will have the older version if it hasn't been updated via the translations app.

Reply:
Ok, so given a translation of a resource how do I, as a project owner, know whether or not that translation needs to be published? I've been looking at some translations on live projects but they all have `links.published_version` set to null. I'm trying to figure out what the logic is, in the front-end, for showing a Publish button to the project owner.

The other question, I guess, is how do I, as a project owner, use the API to publish all pending changes to translations in one go? Publishing each resource individually seems prone to error, in that I might update a workflow, say, but forget that I also need to publish the tutorial.